prisma/seeds/prices.ts

```diff
@@ -1,35 +1,105 @@
-    updatedAt: new Date(),
-    timestamp: 1664593200,
-    networkId: 1,
-    quoteId: 1
-  },
-  { // XECCAD
-    value: new Prisma.Decimal('0.00005663'),
-    createdAt: new Date(),
-    updatedAt: new Date(),
-    timestamp: 1664593200,
-    networkId: 1,
-    quoteId: 2
-  },
-  { // BCHUSD
-    value: new Prisma.Decimal('117.9081'),
-    createdAt: new Date(),
-    updatedAt: new Date(),
-    timestamp: 1664593200,
-    networkId: 2,
-    quoteId: 1
-  },
-  { // BCHCAD
-    value: new Prisma.Decimal('163.0735'),
-    createdAt: new Date(),
-    updatedAt: new Date(),
-    timestamp: 1664593200,
-    networkId: 2,
-    quoteId: 2
+import { Price, Prisma } from '@prisma/client'
+import { getAllPricesByTicker } from 'services/priceService'
+import { PRICE_FILE_MAX_RETRIES, TICKERS, NETWORK_IDS, QUOTE_IDS, KeyValueT } from 'constants/index'
+import { readCsv, fileExists } from 'utils/index'
+import moment from 'moment'
+import * as fs from 'fs'
+import * as path from 'path'
+import { promisify } from 'util'
+
+interface PriceFileData {
+  ticker: string
+  date: string
+  priceInCAD: string
+  priceInUSD: string
+}
+
+export const PATH_PRICE_CSV_FILE = path.join('prisma', 'seeds', 'prices.csv')
+
+async function writeToFile (fileName: string, content: PriceFileData[]): Promise<void> {
+  const writeFile = promisify(fs.writeFile)
+
+  const headers = ['ticker', 'date', 'priceInCAD', 'priceInUSD']
+  const rows = content.map(({ ticker, date, priceInCAD, priceInUSD }) => [ticker, date, priceInCAD, priceInUSD])
+  const csv = [headers, ...rows].map(row => row.join(',')).join('\n')
+
+  await writeFile(fileName, csv, 'utf8')
+}
+
+export async function createPricesFile (): Promise<void> {
+  console.log('Price file creation started')
+  const start = moment()
+  const prices: PriceFileData[] = []
+
+  await Promise.all(Object.values(TICKERS).map(async (ticker) => {
+    const pricesByTicker = await getAllPricesByTicker(ticker)
+
+    pricesByTicker?.forEach(price => {
+      prices.push({
+        ticker,
+        date: price.day,
+        priceInCAD: price.Price_in_CAD,
+        priceInUSD: price.Price_in_USD
+      })
+    })
+  }))
+
+  prices.sort((a, b) => {
+    if (a.ticker < b.ticker) return -1
+    if (a.ticker > b.ticker) return 1
+    if (a.date < b.date) return -1
+    if (a.date > b.date) return 1
+    return 0
+  })
+  await writeToFile(PATH_PRICE_CSV_FILE, prices)
+
+  const finish = moment()
+  console.log(`\n\nstart: ${start.format('HH:mm:ss')}\nfinish: ${finish.format('HH:mm:ss')}\nduration: ${(finish.diff(start) / 1000).toFixed(2)} seconds`)
+}
+
+async function getPricesFromFile (attempt: number = 1): Promise<PriceFileData[]> {
+  if (await fileExists(PATH_PRICE_CSV_FILE)) {
+    const csvContent = await readCsv(PATH_PRICE_CSV_FILE)
+    const res: PriceFileData[] = []
+
+    const headers = csvContent[0]
+    const data = csvContent.slice(1)
+
+    for (const line of data) {
+      const newPrice: KeyValueT<any> = {}
```

Comment:
Changed it to `KeyValueT<string>`, the one you suggested can't be set because all attributes are mandatory and I am initializing it as an empty object here.

About the outdated conversation below about throwing an error if one of the quotes come back missing from the api, I think we should do that when the data comes back from the api and not when reading the file (user can manually change the file and we don't wanna create the file with these wholes then). This was addressed there.